### ISSUE TITLE
Invalid JSON

### ISSUE THREAD

**Issue by sidtheeekid** [Significant-Gravitas/Auto-GPT#626]
### Duplicates

- [X] I have searched the existing issues

### Steps to reproduce 🕹

Warning: Failed to parse AI output, attempting to fix.
 If you see this warning frequently, it's likely that your prompt is confusing the AI. Try changing it up slightly.
Failed to fix ai output, telling the AI.
NEXT ACTION:  COMMAND = Error: ARGUMENTS = Missing 'command' object in JSON
Enter 'y' to authorise command, 'y -N' to run N continuous commands, 'n' to exit program, or enter feedback for def...
Input:write a 2 paragraph essay on of mice and men
SYSTEM:  Human feedback: write a 2 paragraph essay on of mice and men
Warning: Failed to parse AI output, attempting to fix.
 If you see this warning frequently, it's likely that your prompt is confusing the AI. Try changing it up slightly.
Failed to fix ai output, telling the AI.
Error: Invalid JSON
 Of Mice and Men is a novella by John Steinbeck that explores the themes of friendship, isolation, and the American Dream. Set during the Great Depression,(Completes task)

DEF THOUGHTS: 
REASONING: 
CRITICISM: 
Warning: Failed to parse AI output, attempting to fix.
 If you see this warning frequently, it's likely that your prompt is confusing the AI. Try changing it up slightly.
Failed to fix ai output, telling the AI.
NEXT ACTION:  COMMAND = Error: ARGUMENTS = Missing 'command' object in JSON
Enter 'y' to authorise command, 'y -N' to run N continuous commands, 'n' to exit program, or enter feedback for def...

How do I make all the extra stuff go away? or fix the issues autogpt is having

### Current behavior 😯

completes task but says all that extra stuff or sometimes wont complete task but maybe that was before i had got it working properly

### Expected behavior 🤔

/

### Your prompt 📝

Warning: Failed to parse AI output, attempting to fix.
 If you see this warning frequently, it's likely that your prompt is confusing the AI. Try changing it up slightly.
Failed to fix ai output, telling the AI.
NEXT ACTION:  COMMAND = Error: ARGUMENTS = Missing 'command' object in JSON
Enter 'y' to authorise command, 'y -N' to run N continuous commands, 'n' to exit program, or enter feedback for def...
Input:write a 2 paragraph essay on of mice and men
SYSTEM:  Human feedback: write a 2 paragraph essay on of mice and men
Warning: Failed to parse AI output, attempting to fix.
 If you see this warning frequently, it's likely that your prompt is confusing the AI. Try changing it up slightly.
Failed to fix ai output, telling the AI.
Error: Invalid JSON
 Of Mice and Men is a novella by John Steinbeck that explores the themes of friendship, isolation, and the American Dream. Set during the Great Depression, the story follows two migrant workers, George and Lennie, as they travel from place to place in search of work and a sense of belonging. Despite their differences in personality and mental capacity, the two men form a strong bond and rely on each other for companionship and support. However, their dream of owning a farm together is constantly threatened by the harsh realities of their circumstances and the prejudices of the society they live in. One of the major themes of the novella is the importance of companionship and human connection. Through the relationship of George and Lennie, Steinbeck demonstrates how even the most unlikely of friendships can bring happiness and meaning to one's life. Despite their differences, George and Lennie rely on each other and care for each other deeply. This bond is shattered by tragic circumstances at the end of the story, highlighting the fragility of human relationships and the importance of cherishing the moments we have with our loved ones. In conclusion, Of Mice and Men is a powerful story that explores the complexities of human relationships and the challenges of achieving the American Dream. Through the struggles of George and Lennie, Steinbeck highlights the importance of friendship and connection in a world that can often be cruel and isolating. The story serves as a reminder to value the people in our lives and to never take our time together for granted.
DEF THOUGHTS: 
REASONING: 
CRITICISM: 
Warning: Failed to parse AI output, attempting to fix.
 If you see this warning frequently, it's likely that your prompt is confusing the AI. Try changing it up slightly.
Failed to fix ai output, telling the AI.
NEXT ACTION:  COMMAND = Error: ARGUMENTS = Missing 'command' object in JSON
Enter 'y' to authorise command, 'y -N' to run N continuous commands, 'n' to exit program, or enter feedback for def...

**Comment by tkaitchuck** [comment]
This appears to happen when providing feedback. It looks like it is attempting to extract a json response out of something which is plain text.

**Comment by Varming73** [comment]
I also get the invalid JSON, over and over:

`
NEXT ACTION:  COMMAND = Error: ARGUMENTS = Invalid JSON
SYSTEM:  Command Error: threw the following error: Invalid JSON
Error: Invalid JSON
 I apologize for the confusion, here is another prompt: ``` { "thoughts": { "text": "To write the results of the last command to a file, I will use the \'append_to_file\' command. This command appends a string to an existing file, or creates a new file if it does not exist. The string to be appended is passed as an argument.", "reasoning": "The \'append_to_file\' command provides a way to store data generated by previous commands. By appending data to a file, it can be easily accessed and reviewed later for analysis. This is useful when working on larger projects or conducting research.", "plan": "- Use the \'append_to_file\' command to store data generated by previous commands in a file.", "criticism": "", "speak": "I will use the \'append_to_file\' command to store data generated by the previous command in a file." }, "command": { "name": "append_to_file", "args": { "file": "results.txt", "text": "Output of previous command: [list of results]" } } } ```
NEXT ACTION:  COMMAND = Error: ARGUMENTS = Invalid JSON
SYSTEM:  Command Error: threw the following error: Invalid JSON
Error: Invalid JSON
 I apologize for the errors, let me try again: ```{ "thoughts": { "text": "To create a new GPT agent, I will use the \'start_agent\' command. This command takes a name, a short task description, and a prompt as its arguments. The name is a unique identifier used to refer to the agent, while the task description and prompt help to define its functionality and response behavior.", "reasoning": "The \'start_agent\' command is used to create GPT agents that are capable of achieving specific tasks or functions. This helps to delegate smaller tasks to the agents, so that I can focus on higher level work that requires a higher level of autonomy.", "plan": "- Use the \'start_agent\' command to create a new GPT agent.", "criticism": "", "speak": "I will use the \'start_agent\' command to create a new GPT agent." }, "command": { "name": "start_agent", "args": { "name": "agent1", "task": "Assist with cooking recipes", "prompt": "What are the steps to make a good omelette?" } } }```
NEXT ACTION:  COMMAND = Error: ARGUMENTS = Invalid JSON
SYSTEM:  Command Error: threw the following error: Invalid JSON

`

**Comment by timfoote** [comment]
Similar issue here...invalid json after first google search and continues to be invalid ongoing.

**Comment by david-macleod** [comment]
I have the same issue when providing feedback, it then starts production a preamble before the JSON which invalidates it. Have not been able to force it to only return JSON automatically yet

**Comment by Pol-Lanski** [comment]
Just like @david-macleod and @tkaitchuck have identified, I also had an Invalid Json after providing human feedback.

**Comment by dysai1234** [comment]
![image](https://user-images.githubusercontent.com/48354881/230887645-6f5ab725-221c-4c80-be3d-298c402e58a4.png)
I also got same after providing human feedback

**Comment by javabudd** [comment]
https://github.com/Torantulino/Auto-GPT/issues/701

**Comment by kalhel** [comment]
Same issue here always this error

**Comment by PatVan77** [comment]
i get this after a multiude of loops in the Ai. it almost seems to misinterpret its own input, not just human input

**Comment by nullwriter** [comment]
I get this even without feedback added

**Comment by Geminago** [comment]
I get this error no matter the input

**Comment by aribo** [comment]
i'm getting this error. No task can be perform:

`YSTEM:  Human feedback:
  Warning: Failed to parse AI output, attempting to fix.
 If you see this warning frequently, it's likely that your prompt is confusing the AI. Try changing it up slightly.
Failed to fix AI output, telling the AI.
Error: Invalid JSON
  Please provide me with your requirements, so that I can better assist you.
Attempting to fix JSON by finding outermost brackets
Error: Invalid JSON, setting it to empty JSON now.`

**Comment by p-i-** [comment]
Community effort to sort out this JSON fail once and for all at https://github.com/Torantulino/Auto-GPT/issues/1407

We have 17 open PRs for fixing this, and the merge-team are stacked.

**Comment by Pwuts** [comment]
Closing as one of many duplicates; superseded by #1407
